sfge/src/test/java/com/salesforce/rules/unusedmethod/OverloadsTest.java

```diff
@@ -289,4 +290,81 @@ public void callConstructorWithDifferentSignatureOverloads_expectViolation(
                     assertEquals(beginLine, ((MethodVertex) v.getSourceVertex()).getArity());
                 });
     }
+
+    /**
+     * If there's different overloads of a constructor, then only the ones that are actually invoked
```

Comment:
Thinking about this statement some more, we often mark util classes as final and make their constructors private so that the class is never instantiated. In this case, the constructor we mark as private has the same signature as the default constructor. We don't expect any invocation of this private constructor either. 

Do you think we should consider the arity of the private constructor before checking if it's used or not?

Reply:
I guess we could?
On the other hand, they could just use an engine directive for a `util` class's constructor, right?